### PR TITLE
Feature: Add cost multiplier to affect Expenditure types

### PR DESCRIPTION
**Pull request by ltsSmitty**
I've seen various requests to be able to scale pricing up or down, such as making ride building costs 50% more expensive, or landscaping costs 25% cheaper. This could be used to increase difficulty, or to create a new kind of game mode (e.g., a roguelike that influences costs every few minutes to change complexity, etc.) 

This PR goes about it by adding a new cost multiplier expenditure table vs the % to multiply.
```cpp 
/**
* Sets wages to be 50% of their standard value. E.g.handymen are now $25 per month.
**/
gameState.CostMultiplierExpenditureTable[EnumValue(ExpenditureType::Wages)] = 50;
```

A helper function is introduced to interface between the original hardcoded cost and the new cost to be calculated, `money64 FinanceGetModifiedCost(money64 cost, ExpenditureType type)`. 

There are very few places in the codebase where the financial transaction actually occurs, so implementing there was straightforward. The remaining changes are on various UI elements, ensuring that the price shown aligns with the new charge.

This build has a 50% markdown on all costs, implemented in `FinanceInit()`. 

Since this touches so many pieces of code, I'm going to pause here and wait for feedback. Hopefully this can open the door to future fun!

### CODE REVIEW

**Comment by Sadret** [comment]
Where would you set the cost modifier in the UI? Will it only work together with a plugin? If yes, I do not think this is the right way to go. Afaik we have not yet introduced any game mechanics that are only accessible by plugins.

My suggestion would be to solve this by a cost expenditure hook where you can modify the costs, similar how it is done with game actions, I think. (I was always told game actions could be cancelled by changing the result to an error, but I never checked if that worked, especially in multiplayer.)

**Comment by Sadret** [comment]
> My suggestion would be to solve this by a cost expenditure hook where you can modify the costs, similar how it is done with game actions.

See implementation of `action.execute` or https://github.com/OpenRCT2/OpenRCT2/pull/21521 for a working example.

**Comment by Basssiiie** [comment]
I agree with Sadret. I think the implementation can be simplified by implementing the strategy of #21521 also for game actions (probably both query and execute). It gives plugins even more control over the cost and should affect less of the internal code base. 🙂

**Comment by Sadret** [comment]
@ltsSmitty Are you still interested in implementing this? I really do like this feature, I think it makes for some great plugins.

**Comment by ltsSmitty** [comment]
I've got some other projects taking all my time right now, so I'm not able to continue right now. I intend to come back to it in time, or am happy to hand it off to another.

File: src/openrct2/entity/Guest.cpp [hunk 1]
```diff
@@ -2305,16 +2308,18 @@ void Guest::SpendMoney(money64& peep_expend_type, money64 amount, ExpenditureTyp
 {
     assert(!(GetGameState().Park.Flags & PARK_FLAGS_NO_MONEY));
 
-    CashInPocket = std::max(0.00_GBP, static_cast<money64>(CashInPocket) - amount);
-    CashSpent += amount;
+    auto modifiedCost = FinanceGetModifiedCost(amount, expenditure);
```
Comment (Sadret) [inline src/openrct2/entity/Guest.cpp:2311]: I think this is wrong. The modifier should be calculated _after_ the peep gave away its money (but before the player gets the money).

**Pull request closed without merging**